include/deal.II/base/array_view.h

```diff
@@ -305,13 +332,32 @@ namespace internal
 
 
 
+template <typename ElementType, typename MemorySpaceType>
+inline ArrayView<ElementType, MemorySpaceType>::ArrayView()
+  : starting_element(nullptr)
+  , n_elements(0)
+{}
+
+
+
 template <typename ElementType, typename MemorySpaceType>
 inline ArrayView<ElementType, MemorySpaceType>::ArrayView(
   value_type *      starting_element,
   const std::size_t n_elements)
-  : starting_element(starting_element)
-  , n_elements(n_elements)
 {
+  reinit(starting_element, n_elements);
```

Comment:
I would prefer if this constructor still initializes the members instead of calling the member function. The way you do it here, there is a time where the members are uninitialized, and one might trip over that at some point in the future when reworking `reinit()`.